mava/wrappers/jumanji.py

```diff
@@ -265,3 +267,104 @@ def observation_spec(self) -> specs.Spec[Union[Observation, ObservationGlobalSta
             return specs.Spec(ObservationGlobalState, "ObservationSpec", **obs_data)
 
         return specs.Spec(Observation, "ObservationSpec", **obs_data)
+
+
+class CleanerWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the Cleaner environment."""
+
+    def __init__(self, env: Cleaner, add_global_state: bool = False):
+        super().__init__(env, add_global_state)
+        self._env: Cleaner
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Cleaner environment."""
+
+        def create_agents_view(grid: chex.Array, agents_locations: chex.Array) -> chex.Array:
```

Comment:
These comments are amazing, makes it so easy to follow! Awesome work here :fire: